docs/reference/commandline/version.md

```diff
@@ -61,22 +120,74 @@ Server:
   GitCommit:        fec3683
 ```
 
+### API version and version negotiation
+
+The API version that is used by the client depends on the Docker Engine that the
+cli is connecting with. When connecting with the Docker Engine daemon, the client
+and daemon perform API version negotiation, which selects the highest API version
+that is supported by both the CLI and the Docker Engine.
+
+For example, if the CLI is connecting with a Docker 19.03 engine, it downgrades
+to API version 1.40 (which is the API version supported by Docker Engine 19.03):
+
+```console
+$ docker version --format '{{.Client.APIVersion}}'
+
+1.40
+```
+
+Be aware that API version can also be overridden using the `DOCKER_API_VERSION`
+environment variable, which can be useful for debugging purposes, and disables
+API version negotiation. The following example illustrates an environment where
+the `DOCKER_API_VERSION` environment variable is set. Unsetting the environment
+variable removes the override, and re-enables API version negotiation:
+
+```console
+$ env | grep DOCKER_API_VERSION
+DOCKER_API_VERSION=1.39
+
+$ docker version --format '{{.Client.APIVersion}}'
+1.39
+
+$ unset DOCKER_API_VERSION
+$ docker version --format '{{.Client.APIVersion}}'
+1.41
+```
+
+## Examples
+
+### <a name=format></a> Format the output (--format)
+
+The formatting option (`--format`) pretty-prints container output using a Go
+template. Refer to Go's [text/template](https://pkg.go.dev/text/template) package
+documentation for details of the format.
+
+You can use the --format option to obtain specific information about a
+config. The following example command outputs the creation time of the
+config.
+
 ### Get the server version
 
 ```console
 $ docker version --format '{{.Server.Version}}'
 
-19.03.8
+20.10.14
 ```
 
 ### Get the client API version
 
+The following example prints the API version that is used by the client:
+
 ```console
```

Comment:
This is something that probably should be done for various other examples; make sure that examples have a short introduction / description that outlines it, instead of only having a "heading" immediately followed by a command-line example.

(admitted; sometimes the introductions are a bit "boilerplate", so we should find a good balance)

Reply:
We can work on one command that is frequently used and has more information, optimize that, define based on that one what we'd like to improve on there CLI reference and then revisit each page with this in mind.